hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JobSerializerTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.config.CacheSimpleConfig;
+import com.hazelcast.config.SerializerConfig;
+import com.hazelcast.jet.Observable;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.Sources;
+import com.hazelcast.jet.pipeline.test.TestSources;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.jet.core.JobStatus.COMPLETED;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class JobSerializerTest extends SimpleTestInClusterSupport {
+
+    private static final String MAP_NAME = "map";
+    private static final String CACHE_NAME = "cache";
+    private static final String OBSERVABLE_NAME = "observable";
+
+    @BeforeClass
+    public static void beforeClass() {
+        JetConfig config = new JetConfig();
+        config.getHazelcastConfig()
+              .addCacheConfig(new CacheSimpleConfig().setName(CACHE_NAME))
+              .getSerializationConfig()
+              .addSerializerConfig(new SerializerConfig().setTypeClass(Value.class).setClass(ValueSerializer.class));
+        initializeWithClient(1, config, null);
+    }
+
+    @Test
+    public void when_serializerIsRegistered_then_itIsAvailableForLocalMapSourceAndSink() {
```

Comment:
This should be two tests. Or you can do map reading and writing in one pipeline, I think that's fine.